### PR TITLE
add fleet vehicle configuration modules

### PR DESCRIPTION
**Pull request by damdam-s**

### CODE REVIEW

File: fleet_vehicle_configuration/models/fleet_configuration.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+
+
+class FleetVehicleConfiguration(models.Model):
+    _name = "fleet.vehicle.configuration"
+    _description = "Manage vehicle configuration"
+
+    vehicle_id = fields.Many2one("fleet.vehicle")
```
Comment (ivantodorovich) [inline fleet_vehicle_configuration/models/fleet_configuration.py]:
```suggestion
    vehicle_id = fields.Many2one("fleet.vehicle", required=True, ondelete="cascade")
```
maybe?

File: fleet_vehicle_configuration/models/fleet_configuration.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    name = fields.Char(required=True)
+    max_seats = fields.Integer(required=True, default=10)
```
Comment (ivantodorovich) [inline fleet_vehicle_configuration/models/fleet_configuration.py]: Maybe I'm wrong but I seem to recall you can't have required=True on numeric fields cause they're never `False` in psql
If this is true, then you need an `_sql_constraint`

File: fleet_vehicle_configuration/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+This module provides a way to define several vehicule configuration.
+
+Example: car with 5 seats configuration or 7 seats configuration
```
Comment (ivantodorovich) [inline fleet_vehicle_configuration/readme/DESCRIPTION.rst]: I think this module deserves a longer readme.

What the example suggests here seems to be the model IMO (like when you buy a car and it comes in different models / variants).

But in fact the configuration is an internal thing. It's how you alter the car for your company purposes, right?

File: fleet_vehicle_configuration_event/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Add products "allowed" in vehicle configuration.
```
Comment (ivantodorovich) [inline fleet_vehicle_configuration_event/readme/DESCRIPTION.rst]: IMO the module name is too generic for what it does.
This is a very specific requirement, so the module name should be specific too.

Moreover I have no idea how or why one would use this module 😅 
I think a better readme would help

File: fleet_vehicle_configuration_event_specific_counter/models/fleet_configuration.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+class FleetVehicleConfiguration(models.Model):
+    _inherit = "fleet.vehicle.configuration"
+
+    specific_product_id = fields.Many2one(
+        "product.template",
+        domain=[("detailed_type", "=", "event")],
+        string="Specific Event product",
+        help="Event products for the specific max seat counter",
+    )
+    specific_max_seats = fields.Integer(default=0)
```
Comment (ivantodorovich) [inline fleet_vehicle_configuration_event_specific_counter/models/fleet_configuration.py]: "specific" doesn't really mean anything 🤔 
No clue what this module does 😅

**Review by ivantodorovich — COMMENTED** [review]
To ease reviews, it's better to split PRs when multiple modules are added.

File: fleet_vehicle_configuration/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2022 Camptocamp (https://www.camptocamp.com).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+{
+    "name": "Fleet Vehicule Configuration",
```
Comment (marcelsavegnago) [inline fleet_vehicle_configuration/__manifest__.py]:
```suggestion
    "name": "Fleet Vehicle Configuration",
```

**Review by marcelsavegnago — CHANGES_REQUESTED** [review]

File: fleet_vehicle_configuration/models/fleet_configuration.py (hunk 1 above)
Comment (damdam-s) [reply fleet_vehicle_configuration/models/fleet_configuration.py]: I can't set it as required as it works like "order_id" field on sale_order_line

Comment (ivantodorovich) [reply fleet_vehicle_configuration/models/fleet_configuration.py]: What do you mean? `order_id` is required in sale.order.line 😅 

The question is: does it make sense to have `fleet.vehicle.configuration` without an actual `vehicle`? If so, what do they mean?

**Review by ivantodorovich — APPROVED** [review]

**Review by jcoux — APPROVED** [review]

**Pull request merged**